routine.go

```diff
@@ -0,0 +1,159 @@
+package capataz
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/google/uuid"
+)
+
+type Routine interface {
+	Go(context.Context) error
+}
+
+type RoutineFunc func(context.Context) error
+
+func (f RoutineFunc) Go(ctx context.Context) error {
+	return f(ctx)
+}
+
+type RoutineFactory func() (Routine, error)
```

Comment:
I'm still a bit confused about the notion of a Factory here, how are were you envisioning? from the code that is here right now, I'm not able to truly assess its purpose